homeassistant/components/light/zwave.py

```diff
@@ -328,6 +328,10 @@ def turn_on(self, **kwargs):
         elif ATTR_HS_COLOR in kwargs:
             self._hs = kwargs[ATTR_HS_COLOR]
 
+        if ATTR_HS_COLOR in kwargs:
+            # white LED must be off in order for color to work
+            self._white = 0
```

Comment:
Assuming that some devices _do_ allow setting of all the individual RGBW values, It might make sense to only do this zeroing if no `ATTR_WHITE_VALUE` is present?

Reply:
You could merge this branch with the `ATTR_HS_COLOR` branch right before it.

Reply:
@amelchio thanks for taking a look at this.

Good point re: only zeroing if ``ATTR_WHITE_VALUE`` is present. So to be clear, we should be zeroing if ``ATTR_HS_COLOR in kwargs and ATTR_WHITE_VALUE not in kwargs`` ?

Regarding merging with the branch before it, that one is a ``if ATTR_COLOR_TEMP in kwargs`` / ``elif ATTR_HS_COLOR in kwargs`` ... as far as I know my lights don't support color temp, so I didn't merge that because I have no way of testing the behavior...

Reply:
1) yes (and that will be much shorter if you do merge the two HS branches)

2) `color_temp` and `hs_color` are mutually exclusive so you can count on only one being present and thus ignore the `color_temp` case.

Full disclosure, I don't even have these lights so I am just theorizing.

Reply:
Ok cool, thanks so much. I'm just about to run out for a bit, but I'll push a commit with those changes in a few hours.

Reply:
Assuming we're on the same wavelength, I believe this should be addressed in the two commits I just pushed.